src/go/rpk/pkg/auth0/auth0.go

```diff
@@ -101,29 +100,35 @@ func ValidateToken(token string, endpoint Endpoint) error {
 type Client struct {
 	endpoint Endpoint
 
-	httpCl *http.Client
+	httpCl *httpapi.Client
 }
 
 // NewClient initializes and returns a client for talking to auth0.
-func NewClient(endpoint Endpoint) *Client {
+func NewClient(httpCl *http.Client, endpoint Endpoint) *Client {
+	opts := []httpapi.Opt{
+		httpapi.HTTPClient(httpCl),
+		httpapi.Host(endpoint.URL),
+	}
+
+	apiCl := httpapi.NewClient(opts...)
 	cl := &Client{
 		endpoint: endpoint,
-		httpCl:   &http.Client{Timeout: time.Minute},
+		httpCl:   apiCl,
 	}
 
 	// Our auth0 clients are typically extremely short-lived (to issue a single
```

Comment:
This finalizer isn't needed if we accept the client -- we would expect the caller to not handle idle connections itself

However, I don't think we should accept a client here, it's probably annoying for the the caller (ourselves) to setup. httpapi itself initializes a client if none is specified, so we can leave the client out entirely